### DISCUSSION TITLE
Absence of --dump-import as a environment variable makes version updates complex

### DISCUSSION THREAD

**Discussion by bidoubiwa** [meilisearch/product#218] [Feedback & Feature Proposal]
**Describe the bug**
The flag `--dump-import` has no related environment variable ([see doc](https://docs.meilisearch.com/reference/features/configuration.html#import-dump)). This causes the import dump process in environments where meilisearch runs a service to be complex. 

For example, if MeiliSearch runs with `systemctl` the launch command is located in the following file: 
`/etc/systemd/system/meilisearch.service`

In this file the following line is present: 

```
ExecStart=/usr/bin/meilisearch --env production --db-path /var/lib/meilisearch/data.ms
``` 
I could edit the file with a script but it would mean I have to edit it once to add --import-dump flag and once to remove it. 

So this is another solution I used to avoid file parsing, and string manipulation: 

Imagine I want to update my MeiliSearch version and create a dump before hands. This is the current script:
1. create dump
2. Stop the service
3. delete data.ms
4. Download new version of meilisearch
5. launch meilisearch **where it has been downloaded** using `--import-dump` with the dump id of step 1
`./meilisearch --import-dump  --db-path  /var/lib/meilisearch/data.ms`
6. wait for import 
7. Close MeiliSearch local instance 
8. move local meilisearch binary into /usr/bin/meilisearch to override previous version
9. restart meilisearch service
10. delete dump file

It is now updated to the new MeiliSearch version
 
If we had `MEIL_DUMP_IMPORT` the migration would have been as follow: 

1. create dump
2. stop service
3. delete data.ms
4. download new MeiliSearch version
5. move meilisearch binary into /usr/bin/meilisearch to override previous version
6. create MEIL_DUMP_IMPORT with correct dump path
8. restart meilisearch service
9. Remove env variable: `MEILI_DUMP_IMPORT`
10. delete dump file

**MeiliSearch version:** v0.22.0

**Comment by curquiza** [comment]
Feature request and feedback are not in this repo anymore but in the `product` one :) I move it!

**Comment by gmourier** [comment]
Thanks, @bidoubiwa! 

If there was such an environment variable, it would make little sense to point it to a .dump file specifically. On the other hand, in a more general way, we could have an `upgrade` CLI command which could, for example, list the different dumps contained in a folder and display a prompt to choose the desired file in order to import a specific dump if the most recent one is not wanted.

This environment variable allows to do this in an ad-hoc way and seems to make sense to facilitate the precise import of a dump, which seems to be an operation that is not repeated multiple times for the same exact dump.

**Reply by bidoubiwa** [reply]
Thanks @gmourier, I also changed my mind on the risks of having a `--dump-import`. It is indeed not a good idea as you pointed out.

An upgrade CLI command would be very nice!

**Comment by gmourier** [comment]
FYI @bidoubiwa

As we will work on a configuration file for v0.30, we will also consider introducing the following consistency rule.

**All configuration variables must be modifiable via the configuration file, as an env var and as a cli option.**

Sometimes the cloud team asks us env vars that they need that are only accessible via a CLI option when requested. e.g https://github.com/meilisearch/meilisearch/issues/2721

By introducing that consistency rule, we'd like to avoid this kind of back and forth and make everyone's job easier.

**Comment by gmourier** [comment]
Hey 👋

Locking this discussion since it has been released with `v0.30`

📚 https://docs.meilisearch.com/learn/configuration/instance_options.html#import-dump
